File: OnlineChatApp/ClientApp/src/app/models/chat-room.model.ts
import { Conversation } from "./conversation.model";
import { User } from "./user/user.model";

export class ChatRoom{
  id: number;
  is_group: boolean = false;
  name: string;
  created_date: Date;
  chat_room_type: ChatRoomType = ChatRoomType.OneToOne;
  member_list: User[] = [];
  conversation_list: Conversation[] = [];
  user: User;
}



export enum ChatRoomType{
  Group = 1,
  OneToOne = 2
}

File: OnlineChatApp/Models/ViewModels/ChatRoomModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace OnlineChatApp.Models.ViewModels
{
    public class ChatRoomModel
    {
        public long id { get; set; }
        public bool is_group { get; set; } = false;
        public string name { get; set; }
        public DateTime created_date { get; set; }
        public List<ConversationModel> conversation_list { get; set; }
        public List<UserModel> member_list { get; set; }
        public ChatRoomType chat_room_type { get; set; } = ChatRoomType.OneToOne;
        public UserModel user { get; set; }
    }
}
